wipefs: add --all example with glob pattern (#8453)
tldr-pages/tldr: pages/linux/wipefs.md

`sudo wipefs {{/dev/sdX}}`

> - Wipe all available signature types for the device (no recursion into partitions):

Reviewer: **fixup:** When referring to a single entity such as a file or a device, tldr pages use "a specific {{x}}" by convention:

```suggestion
- Wipe all available signature types for a specific device (no recursion into partitions):
```

**Question:** Regarding the change from 'signatures' to 'signature types', I don't understand this change; I would expect that wiping a signature type indicates that we're somehow erasing the concept of a given signature (gpt, mbr etc), and wiping signatures would actually remove the signatures on-disk. Looking at the [wipefs 2.37.85-637cc man page](https://www.man7.org/linux/man-pages/man8/wipefs.8.html) shows that the term "signature type" isn't really used.

Is there something that I'm understanding that makes "signature type" more correct in this context?

Author: > Is there something that I'm understanding that makes "signature type" more correct in this context?

Hi @adrienthebo,  it's a fair question. The reason I made it more verbose from just "all available signatures" to all "available signature types" was that the `--all` means erasing all signatures of any type at the specified block level, but not all, as in recusing into all/deeper block levels and it does not erase "sub-level" block device signatures. I can see how this is unclear. Perhaps it should be worded:

"Wipe all signatures of any type at a specified device block layer (no recursion into partitions)". Does this sound better? Or would you prefer sticking to the original wording given "(no recursion into partitions)" does make it clear enough?

Maybe the man page helps:

> -a, --all
> Erase all available signatures. The set of erased signatures can be restricted with the -t option.

> -t, --types list
> Limit  the  set of printed or erased signatures.  More than one type may be specified in a comma-separated list. The list or individual types can be prefixed with 'no' to specify the types on which no action should be taken.  For more details see mount(8).

My main issue was that I confused all to imply nested/recursive cleanup, but it doesn't. I noticed other people in various forums shared my misconception.

Author: Seeing no further discussion about why or if the signature type matters, I'll leave this as be and resolve the conversation. It it still seems disagreeable, please open a fresh comment and I'll happily take any suggestions on cleaner wording.

> - Wipe all available signature types for a specific device (no recursion into partitions):

Reviewer 2: ```suggestion
- Wipe all available signature types for a specific device with no recursion into partitions:
```

Author: wipefs: cleaner wording about no recursion

> - Wipe all available signatures for specified device level (no recursion to sub-level block layers):

Reviewer 3: Which level? block level?

Author: "block level" is a bit fuzzy now that I think on it again. My understanding is Linux block layers stack up, such that `/dev/sda` would be the full device, then `/dev/sda1` would be the first partition, etc. But technically, perhaps, we should rather say "device and all partitions", given there's other block stuff, like LVM, which doesn't behave this way.

Also, if one does `wipefs /dev/sda*` and there's a partition with an LVM in the partition, I'm guessing wipefs will only clear up the LVM metadata, but not the filesystem metadata with the logical volumes, etc. So there's only so much wipefs does. To truly recurse to all block layers would require a script.

I'll submit another commit to the pull to word this better. My "block layer" wording is too generic.

(Apologies for a tardy reply.)

`sudo wipefs --all {{/dev/sdX}}`

> - Wipd all available signatures for specified device at all block levels using a glob pattern:

Reviewer 3: ```suggestion
- Wipd all available signatures for the specified device at all block levels using a glob pattern:
```

> `sudo wipefs --all {{/dev/sdX*}}`

Reviewer 2: Why does this need an asterisk, surely just giving the disk is enough if we are wiping all.

Feel free to ignore this, if it is not an error.

Reviewer 3: The `*` is required, perhaps this better
```suggestion
`sudo wipefs --all {{/dev/sdX}}*`
```

Author: @CleanMachine1 

> surely just giving the disk is enough if we are wiping all

Fair question. It depends on intention.

- If you simply want to cleanup the device base signatures, yes, but all means "all signatures or signature types" at the block device level, and does not account for any signatures in other layers.
- If however, by "all" you want every signature across multiple layers, the glob is a partial hack to do that.
- E.g. /dev/sda may have two partions /dev/sda1 and /dev/sda2. If you wanted to wipe not just the partition table / label / signature, but also, filesystem signatures (or LVM, whatever is within the partitions), then the glob helps do that.
- Why? Well, besides a long zero erase, later on, if you use any data loss recovery tools that do a full block scan to find signatures, you ideally don't want any stale not-fully-erased signatures that could float in unallocated blocks read by the recovery program. It would just add to the confusion. E.g. imagine a bad program or something corrupts the partition table and it's backup, then the recovery program might scan the entire drive looking for filesystem signatures.
